src/state/persist.rs

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use canonical::{Canon, Sink, Source};
+use canonical_derive::Canon;
+use dusk_hamt::Map;
+use microkelvin::{
+    Backend, BackendCtor, Compound, PersistError, PersistedId, Persistence,
+};
+use std::fs;
+use std::path::Path;
+
+use crate::state::{Contracts, NetworkState};
+use crate::VMError;
+
+/// The [`NetworkStateId`] is the persisted id of the [`NetworkState`]
+#[derive(Canon, Clone, Copy, Debug)]
+pub struct NetworkStateId {
+    origin: PersistedId,
+    head: PersistedId,
+}
+
+impl NetworkStateId {
+    /// Read from the given path a [`NetworkStateId`]
+    pub fn read<P: AsRef<Path>>(path: P) -> Result<Self, VMError> {
+        let buf = fs::read(&path)?;
+        let mut source = Source::new(&buf[..]);
+        let id = NetworkStateId::decode(&mut source)
+            .map_err(VMError::from_store_error)?;
+
+        Ok(id)
+    }
+
+    /// Write to the given path a [`NetworkStateId`]
+    pub fn write<P: AsRef<Path>>(&self, path: P) -> Result<(), VMError> {
+        let mut buf = [0u8; 36 * 2];
```

Comment:
using non-obvious numbers should be explained

Reply:
Agreed, but I'm not even sure this is the best way to obtain the size of a `PersistId` that derive `Canon`. Is there a better way?

Reply:
A comment explaining the number would also suffice imo